components/ai_chat/resources/page/api/bind_conversation.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as Mojom from '../../common/mojom'
+import getAPI from './'
+
+export interface ConversationBindings {
+  conversationHandler: Mojom.ConversationHandlerRemote
+  callbackRouter: Mojom.ConversationUICallbackRouter
+  initialStatePromise?: ReturnType<Mojom.ServiceRemote['bindConversation']>
```

Comment:
From a purist's point of view, it seems a shame to permanently store a copy of this here when it won't be used after it's first received. Data from it is copied immediately to ConversationContext and ActiveChatContext and then updated separately. Perhaps this should be a one-off event emitter that we hope everything will have subscribed to before it's emitted?

Reply:
Yeah, I think an observer would feel quite natural here, and we could just push the latest `ConversationState` through whenever the bound conversation changes. I think this is worth investigating, but I'm not going to block you on  it :smile:

Reply:
I looked in to changing this so that `bindConversation` accepts an observer function, but it still results in storing a copy of the initial data in ActiveChatContext, since ConversationContext is where the state is actually used and modified during the course of app usage. I think this is an example of something we can improve in the current state management discussions.